setup.py

```diff
@@ -1,15 +1,14 @@
 import re
 from functools import partial
 from setuptools import setup, find_packages
-from distutils.util import convert_path
+from pathlib import Path
 
-version = {}
-with open(convert_path('src/version.py')) as f:
-    exec(f.read(), version)
+version_info = re.search(r"[\d.]+", Path("src/version.py").read_text())
+version = {'__version__': version_info[0]}
```

Comment:
```suggestion
version_info = re.search(
    r"""__version__ *= *['"]([^'"]+)['"]""",
    Path("src/version.py").read_text()
).group(1)
version = {'__version__': version_info}
```

[PEP 440](https://peps.python.org/pep-0440/)